sk0-front/src/app/auth/auth.service.ts

```diff
@@ -1,22 +1,23 @@
 import { Injectable } from '@angular/core';
 import { AlertController } from '@ionic/angular';
+import { HttpClient, HttpErrorResponse } from '@angular/common/http';
+import { environment } from 'src/environments/environment';
+import { BehaviorSubject } from 'rxjs';
+import { NavController } from '@ionic/angular';
 
-type User = {
-  name: string;
-  email: string;
-  password: string;
-};
+interface LoginResponse {
+    name: string;
+    email: string;
+    id: number;
+}
 
 @Injectable({
   providedIn: 'root',
 })
 export class AuthService {
-  private _isUserLoggedIn = false;
+  private _isUserLoggedIn = new BehaviorSubject<boolean> (false);
   private _userName = 'Vendég';
```

Comment:
As far as I know we no longer use the underscore as a "prefix to sign its internal usage". It is an antipattern, please avoid it. (there is probably a linter rule for it)

Reply:
Can we set the isUserLoggedIn property public? I might need to access it for the authentication guard (Im not sure yet).

Reply:
@markkovari  isn't that a better solution to have a private value that can be accessed with a getter from outside, not to change "accidentally" from outside of the service?

Reply:
if we don't use underscores or getters, just set the variables to public and that's it?

Reply:
> if we don't use underscores or getters, just set the properties to public and that's it?

It can stay private, Ill just put in a getter function for this property when Im working on my branch. If I got it right from Márk, only the underscore is not necessary.

Reply:
Yes, I was asking about the variable name. We use this convention, when we are not using the variable of the metohd, but here it should be used.

Reply:
If we are following OOP best practices, everything should be private unless you need it to be.